util/tele/span_logger.go

```diff
@@ -88,14 +88,18 @@ func (s *spanLogSink) Error(err error, msg string, keysAndValues ...interface{})
 	)
 }
 
-func (s *spanLogSink) WithValues(keysAndValues ...interface{}) logr.LogSink {
-	s.vals = append(s.vals, keysAndValues...)
-	return s
+func (s spanLogSink) WithValues(keysAndValues ...interface{}) logr.LogSink {
+	// always create a new slice to avoid multiple loggers writing to the same backing array
+	vals := make([]interface{}, len(s.vals)+len(keysAndValues))
+	copy(vals, s.vals)
+	copy(vals[len(s.vals):], keysAndValues)
+	s.vals = vals
```

Comment:
Beyond just the memory issue, I noticed a separate issue where a plain `append` could be clobbering the values in other sibling loggers when the backing array happens to have extra capacity. Replacing all of this with `s.vals = append(s.vals, keysAndValues...)` and running the new unit test should show that issue.